echopype/calibrate/env_params.py

```diff
@@ -59,199 +68,289 @@ def harmonize_env_param_time(
         return p
 
 
-def get_env_params_AZFP(echodata: EchoData, user_env_dict: Optional[dict] = None):
-    """Get env params using user inputs or values from data file.
+def sanitize_user_env_dict(
+    user_dict: Dict[str, Union[int, float, list, xr.DataArray]],
+    channel: Union[List, xr.DataArray],
```

Comment:
Any reason to use `list` in one case and `typing` `List` in the other?

Reply:
Actually, given how frequently you're using the combination of a list and a dataarray, I suggest you define a global variable for that union, then reuse that. It'll make for more compact and consistent code. eg:

```python
sequence = Union[List, xr.DataArray]

# ...

user_dict: Dict[str, Union[int, float, sequence]],
channel: sequence,
```

Reply:
But if we do this, the type hint users get would also be `sequence`, which may not be clear to users?

Reply:
Apparently, I have not finished the test cases I planned to do!! 🤯 Good that you caught this!

Reply:
That's a very good point. The strategy I suggested may not be a net benefit for user-facing functions, for the reason you bring up. But it may be a net benefit for internal functions, where code consistency is a higher priority.

So, scratch that. My only remaining comment here is the original one, which is very minor:

> Any reason to use `list` in one case and `typing` `List` in the other?

Reply:
Oh i forgot the `list` part. I'll change that to `List` in the function typing, for consistency.